apps/meteor/client/components/RoomIcon/OmnichannelRoomIcon/lib/OmnichannelRoomIcon.ts

```diff
@@ -18,20 +18,26 @@ const OmnichannelRoomIcon = new (class extends Emitter {
 		if (this.icons.has(`${appId}-${icon}`)) {
 			return `${appId}-${icon}`;
 		}
-		APIClient.get(`/apps/public/${appId}/get-sidebar-icon`, { icon }).then((response: any) => {
-			this.icons.set(
-				`${appId}-${icon}`,
-				DOMPurify.sanitize(response, {
-					FORBID_ATTR: ['id'],
-					NAMESPACE: 'http://www.w3.org/2000/svg',
-					USE_PROFILES: { svg: true, svgFilters: true },
-				})
-					.replace(`<svg`, `<symbol id="${appId}-${icon}"`)
-					.replace(`</svg>`, '</symbol>'),
-			);
-			this.emit('change');
-			this.emit(`${appId}-${icon}`);
-		});
+		APIClient.send(`/apps/public/${appId}/get-sidebar-icon?icon=${icon}`, 'GET')
```

Comment:
Note: we cannot use .get() now since it is accepting only JSON response and fails if it gets a non-JSON response

Reply:
Is this problem on APIClient side or in apps side? 👀 (I mean, where should this be fixed?)

Reply:
It's a general issue of the REST API, as it was devised as just delivering JSON responses. `APIClient.get()` assumes the response is encoded as one.

But if this endpoint is supposed to respond **only** SVGs, which are text-based, it's fine to encode it as an JSON value, as strings are valid i.e. `JSON.parse("\"<svg>...</svg>\"") === "<svg>...</svg>"` (mind the escaped quotes).

Reply:
So, we can leave the .send for now until we update the apps to encode the svg as JSON 👀 
wdyt of adding a note for it, @murtaza98 ? 👀

Reply:
> So, we can leave the .send for now until we update the apps to encode the svg as JSON eyes wdyt of adding a note for it, @murtaza98 ? eyes

Done :heavy_check_mark: